client.go

```diff
@@ -80,16 +86,24 @@ func (c Client) GetResources() ([]PublicResource, error) {
 	return resources, err
 }
 
-func (c Client) CloneSnap(resource string, snap time.Time) (*PublicClone, error) {
+func (c Client) CloneSnap(resource string, snap time.Time, claimArgs ClaimArgs) (*PublicClone, error) {
 	var snapStr string
 	if !snap.IsZero() {
 		snapStr = snap.Format(utils.TimestampFormat)
 	}
 	uri := strings.TrimRight(fmt.Sprintf("http://%s/resources/%s/snaps/%s", c.server, resource, snapStr), "/")
+	if claimArgs.ClaimPooled {
+		uri = fmt.Sprintf("http://%s/resources/%s/claim", c.server, resource)
+	}
 	req, err := c.newReq("POST", uri, nil)
 	if err != nil {
 		return nil, err
 	}
+	if claimArgs.ClaimPooled && claimArgs.Ttl != 0 {
+		q := req.URL.Query()
+		q.Add("ttl", strconv.FormatInt(claimArgs.Ttl, 10))
+		req.URL.RawQuery = q.Encode()
+	}
```

Comment:
perhaps a bit slimmer?
```suggestion
        q := req.URL.Query()
	if claimArgs.ClaimPooled {
		uri = fmt.Sprintf("http://%s/resources/%s/claim", c.server, resource)
		q.Add("ttl", strconv.FormatInt(claimArgs.Ttl, 10))
	}
	req, err := c.newReq("POST", uri, nil)
	if err != nil {
		return nil, err
	}
	req.URL.RawQuery = q.Encode()
```

Reply:
Sure. I think we'll want to keep the if though, so we skip sending the ttl unless it's specified

Reply:
Is the order correct in the suggestion though? Looks like you'd use req to create q before req was declared?

Reply:
Can't do `q := req.URL.Query()` before `req, err := c.newReq("POST", uri, nil)`, so I guess there is a reason why the code is written like that..